### ISSUE TITLE
oc cluster up --metrics --version=v3.6.0 fails to deploy metrics

### ISSUE THREAD

**Issue by spadgett** [openshift/origin#15586]
After running the following command, the hawkular and cassandra deployments failed to scale for me.

```
$ oc cluster up --metrics --version=v3.6.0
```

I see these events in the openshift-infra namespace:

```
Failed create  
Error creating: pods "hawkular-cassandra-1-" is forbidden: service account openshift-infra/cassandra was not found, retry after the service account is created
```

```
Failed create  
Error creating: pods "hawkular-metrics-" is forbidden: service account openshift-infra/hawkular was not found, retry after the service account is created
```

Deployer logs:

[openshift-ansible-metrics-job-9td61.txt](https://github.com/openshift/origin/files/1190845/openshift-ansible-metrics-job-9td61.txt)

##### Version

oc version
oc v3.6.0-rc.0+d4bbae0-375
kubernetes v1.7.0+695f48a16f
features: Basic-Auth

Server https://127.0.0.1:8443
openshift v3.6.0+c4dd4cf
kubernetes v1.6.1+5115d708d7

@jcantrill @csrwng @mwringe

**Comment by jaenswart** [comment]
getting exactly the same issue trying to deploy the metrics with --metrics in minishift 1.3.1 and openshift 3.6.0

**Comment by ArminMW** [comment]
Also fails with "minishift start --openshift-version=v3.6.0-rc.0 --metrics [...]" instead of v3.6.0. (I think this did not fail 20 days ago, but I am not 100% sure.)

**Comment by mbornoz** [comment]
Trying this "experimental" minishift feature with openshift 3.6.0 and getting the same issue.

$ minishift version
minishift v1.3.1+f4900b07

$ minishift openshift version
openshift v3.6.0+c4dd4cf
kubernetes v1.6.1+5115d708d7
etcd 3.2.1

It seems to work as expected with openshift 3.5.x and minishift 1.3.x

**Comment by csrwng** [comment]
@jcantrill the hawkular and cassandra service accounts are getting created in the default namespace and not in the openshift-infra. That seems to be fixed by this PR -> https://github.com/openshift/openshift-ansible/pull/4977

**Comment by jcantrill** [comment]
@spadgett Is this issue resolved and can be closed?

**Comment by ArminMW** [comment]
I just gave it a try again 

oc cluster up --host-data-dir ~/origin-etcd/ --metrics --logging --service-catalog --version=v3.6.0 --loglevel=5

Good: deployment of Metrics is executing again, three pods are created in namespace openshift-infra: hawkular-cassandra-1, hawkular-metrics and heapster,

Bad: hawkular-metrics and heapster are not successful.

**Comment by csrwng** [comment]
@ArminMW can you get a log of the hawkular-metrics pod?

**Comment by ArminMW** [comment]
I attached the gzip'ped output of `oc logs hawkular-metrics-mljsd`([hawkular-metrics-mljsd.log.gz](https://github.com/openshift/origin/files/1259270/hawkular-metrics-mljsd.log.gz))

**Comment by jcantrill** [comment]
@mwringe @jsanda The cassandra log appears that maybe there was a transient error as its starting but the hawkular log looks like its missing a library or something.  Can you comment regarding what you see?

**Comment by ArminMW** [comment]
I deleted my OpenShift Origin v3.6 cluster and the **hawkular-cassandra**, **hawkular-metrics** & **heapster** Docker images from the local Docker repository. then I did a clean new setup using:

- `oc cluster up --host-data-dir ~/origin-etcd/ --metrics --logging --service-catalog --version=v3.6.0`
- `oc login -u system:admin`
- `oc adm policy add-cluster-role-to-user cluster-admin admin`
- `oc adm policy add-cluster-role-to-group system:openshift:templateservicebroker-client system:unauthenticated system:authenticated`

Then I waited for the **openshift-ansible-metrics-job** to deploy the **hawkular-cassandra**, **hawkular-metrics** and **heapster** pods (after pulling the Docker images). 

I attached a screenshot of the result as shown in the web console and the gzip'ped logs of the deployer-job and the three pods:
![screenshot from 2017-08-29 15-31-41](https://user-images.githubusercontent.com/21081541/29823535-3a4f677e-8ccf-11e7-9323-81fcac4aefa6.png)

[openshift-ansible-metrics-job-3t1pd.log.gz](https://github.com/openshift/origin/files/1260427/openshift-ansible-metrics-job-3t1pd.log.gz)
[hawkular-cassandra-1-cn0qm.log.gz](https://github.com/openshift/origin/files/1260426/hawkular-cassandra-1-cn0qm.log.gz)
[hawkular-metrics-455sc.log.gz](https://github.com/openshift/origin/files/1260425/hawkular-metrics-455sc.log.gz)
[heapster-38st8.log.gz](https://github.com/openshift/origin/files/1260424/heapster-38st8.log.gz)

**Comment by glennodickson** [comment]
Hi 
I have a similar issue in that Heapster logs with errors and has been really troublesome to solve.
Upon a fresh install of Openshift v3.6 the Heapster is not starting fully with the error message in its logs of:

> E0902 18:39:45.071394 1 summary.go:97] error while getting metrics summary from Kubelet localhost(10.0.2.15:10250): request failed - "500 Internal Server Error", response: "Internal Error: failed RootFsInfo: error trying to get filesystem Device for dir /var/lib/origin/openshift.local.volumes: err: could not find device with major: 0, minor: 15 in cached partitions map"

Steps to replicate are:
1.
> oc cluster up  --logging=true --metrics=true --docker-machine=openshift --use-existing-config=true --host-data-dir=/c/data/dev/config/os --host-config-dir=/c/data/dev/config/os 
oc adm policy add-cluster-role-to-user cluster-admin developer
oc adm policy add-cluster-role-to-user cluster-admin admin
oc adm policy add-cluster-role-to-group cluster-reader system:serviceaccounts
2. 
Review the logs from Heapster pod for error message.


Files have been attached that describe :
- oc adm diagnostics (diag.txt)
- oc get all -o json -n openshift-infra (all_details.txt)
- pod logs (multiple pod logs)

Following listed below :
- Indicate version details for Docker, Openshift, oc, virtualbox  (which seems to be mostly the same) and I have included details below.
- List pod descriptions 


```
Version Details:  
================== 
docker version 
================== 
Client:
 Version:      1.12.5
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   7392c3b
 Built:        Fri Dec 16 06:14:34 2016
 OS/Arch:      windows/amd64
--------------------------------------- 
oc version
===================== 
oc v3.6.0+c4dd4cf
kubernetes v1.6.1+5115d708d7
features: Basic-Auth

Server https://192.168.99.100:8443
openshift v3.6.0+c4dd4cf
kubernetes v1.6.1+5115d708d7
--------------------------------------- 
Virtual Box version 
===================== 
5.1.26r117224
--------------------------------------- 
```

```
oc  get pod --namespace=openshift-infra 
======================================= 
NAME                                  READY     STATUS        RESTARTS   AGE
hawkular-cassandra-1-1szz2            1/1       Running       0          5h
hawkular-metrics-5mt9v                0/1       Pending       0          45m
hawkular-metrics-nvc50                0/1       Terminating   0          3h
heapster-s6brs                        0/1       Terminating   0          5h
heapster-x1w8w                        0/1       Running       4          42m
openshift-ansible-metrics-job-g48hz   0/1       Completed     0          5h
```

```
======================================= 
oc  describe pod  hawkular-cassandra-1-h3pst --namespace=openshift-infra 
======================================= 
Name:			hawkular-cassandra-1-h3pst
Namespace:		openshift-infra
Security Policy:	restricted
Node:			localhost/10.0.2.15
Start Time:		Sun, 03 Sep 2017 02:36:10 +0800
Labels:			metrics-infra=hawkular-cassandra
			name=hawkular-cassandra-1
			type=hawkular-cassandra
Annotations:		kubernetes.io/created-by={"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicationController","namespace":"openshift-infra","name":"hawkular-cassandra-1","uid":"791d5da4-900d-11...
			openshift.io/scc=restricted
Status:			Running
IP:			172.17.0.7
Controllers:		ReplicationController/hawkular-cassandra-1
Containers:
  hawkular-cassandra-1:
    Container ID:	docker://278639a0dc06047307b54cd2b9a0beb542012089f13e72ebcd320fd89ab76d89
    Image:		openshift/origin-metrics-cassandra:v3.6.0
    Image ID:		docker-pullable://openshift/origin-metrics-cassandra@sha256:f23734a36e3c13d7bf6cd34bba254031140a66c4d9451ee05327c04f633ad439
    Ports:		9042/TCP, 9160/TCP, 7000/TCP, 7001/TCP
    Command:
      /opt/apache-cassandra/bin/cassandra-docker.sh
      --cluster_name=hawkular-metrics
      --data_volume=/cassandra_data
      --internode_encryption=all
      --require_node_auth=true
      --enable_client_encryption=true
      --require_client_auth=true
    State:		Running
      Started:		Sun, 03 Sep 2017 02:36:16 +0800
    Ready:		True
    Restart Count:	0
    Limits:
      memory:	2G
    Requests:
      memory:	1G
    Readiness:	exec [/opt/apache-cassandra/bin/cassandra-docker-ready.sh] delay=0s timeout=1s period=10s #success=1 #failure=3
    Environment:
      CASSANDRA_MASTER:			true
      CASSANDRA_DATA_VOLUME:		/cassandra_data
      JVM_OPTS:				-Dcassandra.commitlog.ignorereplayerrors=true
      TRUSTSTORE_NODES_AUTHORITIES:	/hawkular-cassandra-certs/tls.peer.truststore.crt
      TRUSTSTORE_CLIENT_AUTHORITIES:	/hawkular-cassandra-certs/tls.client.truststore.crt
      POD_NAMESPACE:			openshift-infra (v1:metadata.namespace)
      MEMORY_LIMIT:			2000000000 (limits.memory)
      CPU_LIMIT:			node allocatable (limits.cpu)
    Mounts:
      /cassandra_data from cassandra-data (rw)
      /hawkular-cassandra-certs from hawkular-cassandra-certs (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from cassandra-token-pk4tf (ro)
Conditions:
  Type		Status
  Initialized 	True 
  Ready 	True 
  PodScheduled 	True 
Volumes:
  cassandra-data:
    Type:	EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:	
  hawkular-cassandra-certs:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	hawkular-cassandra-certs
    Optional:	false
  cassandra-token-pk4tf:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	cassandra-token-pk4tf
    Optional:	false
QoS Class:	Burstable
Node-Selectors:	<none>
Tolerations:	<none>
Events:
  FirstSeen	LastSeen	Count	From			SubObjectPath				Type		Reason		Message
  ---------	--------	-----	----			-------------				--------	------		-------
  14m		14m		1	default-scheduler						Normal		Scheduled	Successfully assigned hawkular-cassandra-1-h3pst to localhost
  14m		14m		1	kubelet, localhost	spec.containers{hawkular-cassandra-1}	Normal		Pulling		pulling image "openshift/origin-metrics-cassandra:v3.6.0"
  14m		14m		1	kubelet, localhost	spec.containers{hawkular-cassandra-1}	Normal		Pulled		Successfully pulled image "openshift/origin-metrics-cassandra:v3.6.0"
  14m		14m		1	kubelet, localhost	spec.containers{hawkular-cassandra-1}	Normal		Created		Created container
  14m		14m		1	kubelet, localhost	spec.containers{hawkular-cassandra-1}	Normal		Started		Started container
  14m		14m		3	kubelet, localhost	spec.containers{hawkular-cassandra-1}	Warning		Unhealthy	Readiness probe failed: Could not get the Cassandra status. This may mean that the Cassandra instance is not up yet. Will try again
nodetool: Failed to connect to '127.0.0.1:7199' - ConnectException: 'Connection refused (Connection refused)'.

  13m	13m	2	kubelet, localhost	spec.containers{hawkular-cassandra-1}	Warning	Unhealthy	Readiness probe failed: Could not get the Cassandra status. This may mean that the Cassandra instance is not up yet. Will try again
```

```
======================================= 
oc  describe pod  hawkular-metrics-f7kb6 --namespace=openshift-infra 
======================================= 
Name:			hawkular-metrics-f7kb6
Namespace:		openshift-infra
Security Policy:	restricted
Node:			localhost/10.0.2.15
Start Time:		Sun, 03 Sep 2017 02:36:15 +0800
Labels:			metrics-infra=hawkular-metrics
			name=hawkular-metrics
Annotations:		kubernetes.io/created-by={"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicationController","namespace":"openshift-infra","name":"hawkular-metrics","uid":"6f56b8e9-900d-11e7-8...
			openshift.io/scc=restricted
Status:			Running
IP:			172.17.0.9
Controllers:		ReplicationController/hawkular-metrics
Containers:
  hawkular-metrics:
    Container ID:	docker://6520b75d0bfe51342f3173852f9a0f97af59f7a73f14d4b3beeae1b6add5b885
    Image:		openshift/origin-metrics-hawkular-metrics:v3.6.0
    Image ID:		docker-pullable://openshift/origin-metrics-hawkular-metrics@sha256:7eed6704bfd27a282c32a0f5a11e36cf19268633313a5c076ad41c6b6ec7ebbb
    Ports:		8080/TCP, 8443/TCP, 8888/TCP
    Command:
      /opt/hawkular/scripts/hawkular-metrics-wrapper.sh
      -b
      0.0.0.0
      -Dhawkular.metrics.cassandra.nodes=hawkular-cassandra
      -Dhawkular.metrics.cassandra.use-ssl
      -Dhawkular.metrics.openshift.auth-methods=openshift-oauth,htpasswd
      -Dhawkular.metrics.openshift.htpasswd-file=/hawkular-account/hawkular-metrics.htpasswd
      -Dhawkular.metrics.allowed-cors-access-control-allow-headers=authorization
      -Dhawkular.metrics.default-ttl=7
      -Dhawkular.metrics.admin-tenant=_hawkular_admin
      -Dhawkular-alerts.cassandra-nodes=hawkular-cassandra
      -Dhawkular-alerts.cassandra-use-ssl
      -Dhawkular.alerts.openshift.auth-methods=openshift-oauth,htpasswd
      -Dhawkular.alerts.openshift.htpasswd-file=/hawkular-account/hawkular-metrics.htpasswd
      -Dhawkular.alerts.allowed-cors-access-control-allow-headers=authorization
      -Dorg.apache.tomcat.util.buf.UDecoder.ALLOW_ENCODED_SLASH=true
      -Dorg.apache.catalina.connector.CoyoteAdapter.ALLOW_BACKSLASH=true
      -Dcom.datastax.driver.FORCE_NIO=true
      -DKUBERNETES_MASTER_URL=https://kubernetes.default.svc
      -DUSER_WRITE_ACCESS=False
    State:		Running
      Started:		Sun, 03 Sep 2017 02:38:13 +0800
    Last State:		Terminated
      Reason:		Error
      Exit Code:	137
      Started:		Sun, 03 Sep 2017 02:37:14 +0800
      Finished:		Sun, 03 Sep 2017 02:37:56 +0800
    Ready:		True
    Restart Count:	2
    Limits:
      memory:	2500M
    Requests:
      memory:	1500M
    Liveness:	exec [/opt/hawkular/scripts/hawkular-metrics-liveness.py] delay=0s timeout=1s period=10s #success=1 #failure=3
    Readiness:	exec [/opt/hawkular/scripts/hawkular-metrics-readiness.py] delay=0s timeout=1s period=10s #success=1 #failure=3
    Environment:
      POD_NAMESPACE:			openshift-infra (v1:metadata.namespace)
      MASTER_URL:			https://kubernetes.default.svc
      JGROUPS_PASSWORD:			hSBojuDS33P4QYF5G
      TRUSTSTORE_AUTHORITIES:		/hawkular-metrics-certs/tls.truststore.crt
      OPENSHIFT_KUBE_PING_NAMESPACE:	openshift-infra (v1:metadata.namespace)
      OPENSHIFT_KUBE_PING_LABELS:	metrics-infra=hawkular-metrics,name=hawkular-metrics
      STARTUP_TIMEOUT:			500
    Mounts:
      /hawkular-account from hawkular-metrics-account (rw)
      /hawkular-metrics-certs from hawkular-metrics-certs (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from hawkular-token-dmgcb (ro)
Conditions:
  Type		Status
  Initialized 	True 
  Ready 	True 
  PodScheduled 	True 
Volumes:
  hawkular-metrics-certs:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	hawkular-metrics-certs
    Optional:	false
  hawkular-metrics-account:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	hawkular-metrics-account
    Optional:	false
  hawkular-token-dmgcb:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	hawkular-token-dmgcb
    Optional:	false
QoS Class:	Burstable
Node-Selectors:	<none>
Tolerations:	<none>
Events:
  FirstSeen	LastSeen	Count	From			SubObjectPath				Type		Reason		Message
  ---------	--------	-----	----			-------------				--------	------		-------
  14m		14m		1	default-scheduler						Normal		Scheduled	Successfully assigned hawkular-metrics-f7kb6 to localhost
  13m		13m		1	kubelet, localhost	spec.containers{hawkular-metrics}	Normal		Killing		Killing container with id docker://hawkular-metrics:pod "hawkular-metrics-f7kb6_openshift-infra(9a861587-900d-11e7-8032-0a877c2fc508)" container "hawkular-metrics" is unhealthy, it will be killed and re-created.
  14m		13m		2	kubelet, localhost	spec.containers{hawkular-metrics}	Normal		Pulling		pulling image "openshift/origin-metrics-hawkular-metrics:v3.6.0"
  14m		13m		2	kubelet, localhost	spec.containers{hawkular-metrics}	Normal		Pulled		Successfully pulled image "openshift/origin-metrics-hawkular-metrics:v3.6.0"
  14m		13m		2	kubelet, localhost	spec.containers{hawkular-metrics}	Normal		Created		Created container
  14m		13m		2	kubelet, localhost	spec.containers{hawkular-metrics}	Normal		Started		Started container
  14m		13m		5	kubelet, localhost	spec.containers{hawkular-metrics}	Warning		Unhealthy	Liveness probe failed: Failed to access the status endpoint : <urlopen error [Errno 99] Cannot assign requested address>.
Traceback (most recent call last):
  File "/opt/hawkular/scripts/hawkular-metrics-liveness.py", line 48, in <module>
    if int(uptime) < int(timeout):
ValueError: invalid literal for int() with base 10: ''

  14m	13m	6	kubelet, localhost	spec.containers{hawkular-metrics}	Warning	Unhealthy	Readiness probe failed: Failed to access the status endpoint : <urlopen error [Errno 99] Cannot assign requested address>. This may be due to Hawkular Metrics not being ready yet. Will try again.

  13m	12m	2	kubelet, localhost						Warning	FailedSync	Error syncing pod
  13m	12m	3	kubelet, localhost	spec.containers{hawkular-metrics}	Warning	Unhealthy	Readiness probe failed: Failed to access the status endpoint : HTTP Error 404: Not Found. This may be due to Hawkular Metrics not being ready yet. Will try again.
```

```
======================================= 
oc  describe pod  heapster-4x983 --namespace=openshift-infra 
======================================= 
Name:			heapster-4x983
Namespace:		openshift-infra
Security Policy:	restricted
Node:			localhost/10.0.2.15
Start Time:		Sun, 03 Sep 2017 02:36:37 +0800
Labels:			metrics-infra=heapster
			name=heapster
Annotations:		kubernetes.io/created-by={"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicationController","namespace":"openshift-infra","name":"heapster","uid":"860c6abd-900d-11e7-8032-0a87...
			openshift.io/scc=restricted
Status:			Running
IP:			172.17.0.2
Controllers:		ReplicationController/heapster
Containers:
  heapster:
    Container ID:	docker://9c856cfa1315b3826a14e1b881cd95537023152d8d0eac1a63c881d0487c2e39
    Image:		openshift/origin-metrics-heapster:v3.6.0
    Image ID:		docker-pullable://openshift/origin-metrics-heapster@sha256:435255a96935ec61e4a368d1fb70c57a898c8a71bc7b86cfe1dd2b313184ec44
    Port:		8082/TCP
    Command:
      heapster-wrapper.sh
      --wrapper.allowed_users_file=/secrets/heapster.allowed-users
      --source=kubernetes.summary_api:${MASTER_URL}?useServiceAccount=true&kubeletHttps=true&kubeletPort=10250
      --tls_cert=/heapster-certs/tls.crt
      --tls_key=/heapster-certs/tls.key
      --tls_client_ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
      --allowed_users=%allowed_users%
      --metric_resolution=10s
      --wrapper.username_file=/hawkular-account/hawkular-metrics.username
      --wrapper.password_file=/hawkular-account/hawkular-metrics.password
      --wrapper.endpoint_check=https://hawkular-metrics:443/hawkular/metrics/status
      --sink=hawkular:https://hawkular-metrics:443?tenant=_system&labelToTenant=pod_namespace&labelNodeId=nodename&caCert=/hawkular-metrics-certs/tls.crt&user=%username%&pass=%password%&filter=label(container_name:^system.slice.*|^user.slice)
    State:		Running
      Started:		Sun, 03 Sep 2017 02:36:43 +0800
    Ready:		True
    Restart Count:	0
    Limits:
      memory:	3750M
    Requests:
      memory:	937500k
    Readiness:	exec [/opt/heapster-readiness.sh] delay=0s timeout=1s period=10s #success=1 #failure=3
    Environment:
      STARTUP_TIMEOUT:	500
    Mounts:
      /hawkular-account from hawkular-metrics-account (rw)
      /hawkular-metrics-certs from hawkular-metrics-certs (rw)
      /heapster-certs from heapster-certs (rw)
      /secrets from heapster-secrets (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from heapster-token-mtz48 (ro)
Conditions:
  Type		Status
  Initialized 	True 
  Ready 	True 
  PodScheduled 	True 
Volumes:
  heapster-secrets:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	heapster-secrets
    Optional:	false
  heapster-certs:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	heapster-certs
    Optional:	false
  hawkular-metrics-certs:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	hawkular-metrics-certs
    Optional:	false
  hawkular-metrics-account:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	hawkular-metrics-account
    Optional:	false
  heapster-token-mtz48:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	heapster-token-mtz48
    Optional:	false
QoS Class:	Burstable
Node-Selectors:	<none>
Tolerations:	<none>
Events:
  FirstSeen	LastSeen	Count	From			SubObjectPath			Type		Reason			Message
  ---------	--------	-----	----			-------------			--------	------			-------
  14m		14m		5	default-scheduler					Warning		FailedScheduling	No nodes are available that match all of the following predicates:: Insufficient memory (1).
  14m		14m		1	default-scheduler					Normal		Scheduled		Successfully assigned heapster-4x983 to localhost
  14m		14m		1	kubelet, localhost	spec.containers{heapster}	Normal		Pulling			pulling image "openshift/origin-metrics-heapster:v3.6.0"
  14m		14m		1	kubelet, localhost	spec.containers{heapster}	Normal		Pulled			Successfully pulled image "openshift/origin-metrics-heapster:v3.6.0"
  14m		14m		1	kubelet, localhost	spec.containers{heapster}	Normal		Created			Created container
  14m		14m		1	kubelet, localhost	spec.containers{heapster}	Normal		Started			Started container
  14m		11m		17	kubelet, localhost	spec.containers{heapster}	Warning		Unhealthy		Readiness probe failed: The heapster process is not yet started, it is waiting for the Hawkular Metrics to start.
```

```
======================================= 
oc  describe pod  openshift-ansible-metrics-job-f3vhs --namespace=openshift-infra 
======================================= 
Name:			openshift-ansible-metrics-job-f3vhs
Namespace:		openshift-infra
Security Policy:	privileged
Node:			localhost/10.0.2.15
Start Time:		Sun, 03 Sep 2017 02:32:19 +0800
Labels:			controller-uid=0930aaaa-900d-11e7-8032-0a877c2fc508
			job-name=openshift-ansible-metrics-job
Annotations:		kubernetes.io/created-by={"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"Job","namespace":"openshift-infra","name":"openshift-ansible-metrics-job","uid":"0930aaaa-900d-11e7-8032-0...
			openshift.io/scc=privileged
Status:			Succeeded
IP:			10.0.2.15
Controllers:		Job/openshift-ansible-metrics-job
Containers:
  openshift-ansible-metrics-job:
    Container ID:	docker://a2e8e99cbcc5b048b56bfb565cfa233436460f086d4d8d2cea4042d3d05c0e59
    Image:		openshift/origin-ansible:v3.6.0
    Image ID:		docker-pullable://openshift/origin-ansible@sha256:f3aa3bdeb07a3020894106fff82dde1444bb5b5834594fdfb787ea7117c78d72
    Port:		
    State:		Terminated
      Reason:		Completed
      Exit Code:	0
      Started:		Sun, 03 Sep 2017 02:32:20 +0800
      Finished:		Sun, 03 Sep 2017 02:36:29 +0800
    Ready:		False
    Restart Count:	0
    Environment:
      INVENTORY_FILE:	/tmp/inventory
      PLAYBOOK_FILE:	playbooks/byo/openshift-cluster/openshift-metrics.yml
    Mounts:
      /etc/origin/master from configdir (rw)
      /tmp/inventory from inventoryfile (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from openshift-ansible-token-hcs4d (ro)
Conditions:
  Type		Status
  Initialized 	True 
  Ready 	False 
  PodScheduled 	True 
Volumes:
  configdir:
    Type:	HostPath (bare host directory volume)
    Path:	/c/data/dev/config/os/master
  inventoryfile:
    Type:	HostPath (bare host directory volume)
    Path:	/c/data/dev/config/os/metrics.inventory
  openshift-ansible-token-hcs4d:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	openshift-ansible-token-hcs4d
    Optional:	false
QoS Class:	BestEffort
Node-Selectors:	<none>
Tolerations:	<none>
Events:
  FirstSeen	LastSeen	Count	From			SubObjectPath					Type		Reason			Message
  ---------	--------	-----	----			-------------					--------	------			-------
  18m		18m		4	default-scheduler							Warning		FailedScheduling	no nodes available to schedule pods
  18m		18m		1	default-scheduler							Normal		Scheduled		Successfully assigned openshift-ansible-metrics-job-f3vhs to localhost
  18m		18m		1	kubelet, localhost	spec.containers{openshift-ansible-metrics-job}	Normal		Pulled			Container image "openshift/origin-ansible:v3.6.0" already present on machine
  18m		18m		1	kubelet, localhost	spec.containers{openshift-ansible-metrics-job}	Normal		Created			Created container
  18m		18m		1	kubelet, localhost	spec.containers{openshift-ansible-metrics-job}	Normal		Started			Started container
  12m		12m		1	kubelet, localhost							Warning		FailedMount		Unable to mount volumes for pod "openshift-ansible-metrics-job-f3vhs_openshift-infra(093344a9-900d-11e7-8032-0a877c2fc508)": timeout expired waiting for volumes to attach/mount for pod "openshift-infra"/"openshift-ansible-metrics-job-f3vhs". list of unattached/unmounted volumes=[configdir inventoryfile openshift-ansible-token-hcs4d]
  12m		12m		1	kubelet, localhost							Warning		FailedSync		Error syncing pod

```

**Any hints to resolve this would be very appreciated.** 


[all_details.txt](https://github.com/openshift/origin/files/1272437/all_details.txt)
[diag_details.txt](https://github.com/openshift/origin/files/1272438/diag_details.txt)
[hawkular-cassandra-1-h3pst-log.zip](https://github.com/openshift/origin/files/1272439/hawkular-cassandra-1-h3pst-log.zip)
[hawkular-metrics-f7kb6-log.zip](https://github.com/openshift/origin/files/1272435/hawkular-metrics-f7kb6-log.zip)
[heapster-4x983-log.zip](https://github.com/openshift/origin/files/1272434/heapster-4x983-log.zip)
[openshift-ansible-metrics-job-f3vhs-log.zip](https://github.com/openshift/origin/files/1272436/openshift-ansible-metrics-job-f3vhs-log.zip)

**Comment by ArminMW** [comment]
After trying multiple combinations of different versions of the origin-ansible and metrics images, I finally found one combination that works. I managed to get **Metrics v3.6.0-alpha.2** running on an **Origin v3.6.0+c4dd4cf** "oc cluster up" OpenShift cluster.

Before, with "oc cluster up" I was able to 
- either create an v1.5.1 Cluster with working Metrics but broken Elastic/Kibana (logging), 
- or an v3.6.0 Cluster with working Elastic/Kibana (logging) and broken Metrics. 

Now I have an v3.6.0 Cluster with both working, thus working Metrics and working Elastic/Kibana.

How to reproduce:

`oc cluster up --host-data-dir ~/origin-etcd/ --metrics --logging --service-catalog --version=v3.6.0 `

This creates the Cluster with working logging (Elastic/Kibana), Service Catalog, but broken Metrics. Then I manually deleted everything in openshift-infra and redeployed the older version of the Metrics as described here:

I deployed the `openshift/origin-ansible:v3.6.0` image which uses the `/var/lib/origin/openshift.local.config/metrics.inventory` file, in which I modified the `openshift_metrics_image_version` variable and added the `[etcd]` section:
```
[OSEv3:children]
masters
nodes

[OSEv3:vars]
#openshift_release=v3.6.0

openshift_deployment_type=origin 

openshift_metrics_install_metrics=True
openshift_metrics_image_prefix=openshift/origin-
openshift_metrics_image_version=v3.6.0-alpha.2
openshift_metrics_resolution=10s

openshift_metrics_hawkular_hostname=hawkular-metrics-openshift-infra.127.0.0.1.nip.io

[masters]
127.0.0.1 ansible_connection=local

[nodes]
127.0.0.1

[etcd]
127.0.0.1
```
I started the container with the metrics deployer by `oc create -f origin-ansible-deploy.yaml` where the contents of this YAML file is as follows:
```
apiVersion: v1
kind: Pod
metadata:
  annotations:
    openshift.io/scc: privileged
  generateName: openshift-ansible-metrics-job-
  labels:
    job-name: openshift-ansible-metrics-job
  namespace: openshift-infra
spec:
  containers:
    - env:
        - name: INVENTORY_FILE
          value: /tmp/inventory
        - name: PLAYBOOK_FILE
          value: playbooks/byo/openshift-cluster/openshift-metrics.yml
      image: 'openshift/origin-ansible:v3.6.0'
      imagePullPolicy: IfNotPresent
      name: openshift-ansible-metrics-job
      resources: {}
      securityContext:
        privileged: false
        runAsUser: 0
      terminationMessagePath: /dev/termination-log
      terminationMessagePolicy: File
      volumeMounts:
        - mountPath: /etc/origin/master
          name: configdir
        - mountPath: /tmp/inventory
          name: inventoryfile
        - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
          name: openshift-ansible-token-h0r7f
          readOnly: true
  dnsPolicy: ClusterFirst
  hostNetwork: true
  nodeName: localhost
  restartPolicy: Never
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: openshift-ansible
  serviceAccountName: openshift-ansible
  terminationGracePeriodSeconds: 30
  volumes:
    - hostPath:
        path: /var/lib/origin/openshift.local.config/master
      name: configdir
    - hostPath:
        path: /var/lib/origin/openshift.local.config/metrics.inventory
      name: inventoryfile
    - name: openshift-ansible-token-h0r7f
      secret:
        defaultMode: 420
        secretName: openshift-ansible-token-h0r7f
```
For further analysis by the experts, I attached the output of the ansible deployer pod that I extracted with `oc logs po/openshift-ansible-metrics-job-h9x3v`, and the same for the Cassandra, Metrics and Heapster pods:

[hawkular-cassandra-1-9jcgx.log.gz](https://github.com/openshift/origin/files/1287717/hawkular-cassandra-1-9jcgx.log.gz)
[hawkular-metrics-8fc44.log.gz](https://github.com/openshift/origin/files/1287719/hawkular-metrics-8fc44.log.gz)
[heapster-q3xms.log.gz](https://github.com/openshift/origin/files/1287718/heapster-q3xms.log.gz)
[openshift-ansible-metrics-job-h9x3v.log.gz](https://github.com/openshift/origin/files/1287713/openshift-ansible-metrics-job-h9x3v.log.gz)

**Comment by jcantrill** [comment]
Tested with on centos host:
```
# ./oc version
oc v3.6.0+c4dd4cf
kubernetes v1.6.1+5115d708d7
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://127.0.0.1:8443
openshift v3.6.0+c4dd4cf
kubernetes v1.6.1+5115d708d7
```
Results in:
```
# oc get pods
NAME                                  READY     STATUS      RESTARTS   AGE
hawkular-cassandra-1-75dvr            1/1       Running     0          14m
hawkular-metrics-59f47                1/1       Running     0          14m
heapster-7c4rx                        1/1       Running     0          14m
openshift-ansible-metrics-job-41qcg   0/1       Completed   0          16m
```
Looks to work with the latest version of openshift/origin-ansible:v3.6.0
Closing

**Comment by ArminMW** [comment]
I also successfully deployed v3.6.0 "oc cluster up --metrics [...]".

Btw., I noticed that some problems with hawkular/heapster were caused by my node (Ubuntu 16.04 on a laptop) changes its IP-address when moving between different WLANs. The problem seems to be related with certificates. I saw some related messages in the heapster logs. 

Unfortunately the "redeploy node certificates" ansible playbook does not work on an "oc cluster" installation, and I did not succeed to redeploy the certificates manually.

To avoid these problems --- OpenShift/Kubernetes on nodes with non-permanent IP addresses is not really a feature to expect --- I finally decided to stop installing "oc cluster" directly on my laptop (which changes WLANs and IP address frequently) and installed a KVM with CentOS 7.4 and a fixed virtual IP address for my "oc cluster".
